pkg/acs/calacs/lib/mkspt.c

```diff
@@ -191,7 +229,50 @@ int mkNewSpt (char *in_list, char *mtype, char *output) {
         freeHdr (&header);
     }
 
+    /* If necessary, re-open the primary header to update the NEXTEND keyword */
+    if ((extnum != numSPT) || (numSPT != nimgs)) {
+
+        imUpdate = openUpdateImage (out_spt, "", 0, &header);
+        if (hstio_err()) {
+            trlopenerr (out_spt);
+            freeHdr (&header);
+            return (status = OPEN_FAILED);
+        }
+
+        if (PutKeyInt (&header, "NEXTEND", extnum, "")) {
+            reportErrorAndFreeHeader (&header);
+            closeImage (imUpdate);
+            return (status = hstio_err());
+        }
+
+        /* Write the updated header and clean up */
+        /* Ignore the returned value from putHeader as an hstio error is set at the same time */
+        if (putHeader(imUpdate)) {
+            reportErrorAndFreeHeader (&header);
+            closeImage (imUpdate);
+            return (status = hstio_err());
+        }
+
+        closeImage (imUpdate);
+        freeHdr(&header);
+
+        sprintf(MsgText,"Updated output _spt.fits file to reflect %d extensions...\n",extnum);
+        trlmessage(MsgText);
+    }
+
     c_imtclose (tpin);
 	/* Successful return */
 	return (status);
 }
+
+// Simple helper routine to report the HSTIO error to the .tra file and free 
+// the corresponding header 
+static void 
+reportErrorAndFreeHeader (Hdr *hdr)
+{
+	sprintf (MsgText, hstio_errmsg());
```

Comment:
Is this even correct? Using `hstio_errmsg()` instead of the `int status` returned from the offending `PutXXX()` calls? I think (haven't properly checked) you should be using `WhichError(int status)`.